instrumentation/opentelemetry-instrumentation-pulsar/src/opentelemetry/instrumentation/pulsar/utils.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional
+
+import pulsar
+
+from opentelemetry.semconv.trace import (
+    MessagingDestinationKindValues,
+    MessagingOperationValues,
+    SpanAttributes,
+)
+
+
+def _enrich_span(
+    span,
+    topic,
+    operation: Optional[MessagingOperationValues] = None,
+    partition_key=None,
+    **_kwargs,
+):
+    if not span.is_recording():
+        return
+
+    span.set_attribute(SpanAttributes.MESSAGING_SYSTEM, "pulsar")
+    span.set_attribute(SpanAttributes.MESSAGING_DESTINATION, topic)
+
+    if partition_key:
+        span.set_attribute(
+            SpanAttributes.MESSAGING_KAFKA_MESSAGE_KEY, partition_key
```

Comment:
```suggestion
            SpanAttributes.MESSAGING_PULSAR_MESSAGE_KEY, partition_key
```

Reply:
resolving conversation as `MESSAGING_PULSAR_MESSAGE_KEY` does not exist and it means that the semantic convention would have to add it